lib/widgets/share.dart

```diff
@@ -171,24 +199,88 @@ class SharePage extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    final zulipLocalizations = ZulipLocalizations.of(context);
+    final globalStore = GlobalStoreWidget.of(context);
+    final store = PerAccountStoreWidget.of(context);
     final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    final hasMultipleAccounts = globalStore.accountIds.length > 1;
+
+    Widget mkTabLabel({required String text, required IconData icon}) {
+      return ConstrainedBox(
+        constraints: BoxConstraints(minHeight: 42),
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          spacing: 4,
+          children: [
+            Icon(size: 24, icon),
+            Flexible(
+              child: Text(
+                text,
+                overflow: TextOverflow.ellipsis,
+                maxLines: 1,
+                style: TextStyle(
+                  fontSize: 18,
+                  height: 24 / 18,
+                ).merge(weightVariableTextStyle(context, wght: 500)))),
+          ]));
+    }
 
     return DefaultTabController(
       length: 2,
-      child: Scaffold(
-        appBar: AppBar(
-          title: Text(zulipLocalizations.sharePageTitle),
-          bottom: TabBar(
-            indicatorColor: designVariables.icon,
-            labelColor: designVariables.foreground,
-            unselectedLabelColor: designVariables.foreground.withFadedAlpha(0.7),
+      child: Column(children: [
+        Row(children: [
+          GestureDetector(
+            behavior: HitTestBehavior.opaque,
+            onTap: !hasMultipleAccounts
+              ? null
+              : () {
+                  ChooseAccountForShareDialog.show(
+                    pageContext: context,
+                    selectedAccountId: store.accountId,
+                    sharedFiles: sharedFiles,
+                    sharedText: sharedText);
+                },
+            child: Padding(
+              padding: EdgeInsets.symmetric(vertical: 7, horizontal: 11),
+              child: AvatarShape(
+                size: 28,
+                borderRadius: 4,
+                child: RealmContentNetworkImage(
+                  store.resolvedRealmIcon,
+                  filterQuality: FilterQuality.medium,
+                  fit: BoxFit.cover)))),
+          Expanded(child: TabBar(
+            padding: EdgeInsets.zero,
```

Comment:
```suggestion
          Expanded(child: TabBar(
```

This is equivalent, right?